test/unit/modules/blocks/utils.js

```diff
@@ -13,268 +13,558 @@
+			var blockObject = blocksUtilsModule.readDbRows(viewRow_full_blocks_list);
+
+			expect(blockObject).to.be.an('array');
+			expect(blockObject[0]).to.be.an('object');
+			expect(blockObject[0].id).to.equal('13068833527549895884');
+			expect(blockObject[0].height).to.equal(3);
+			expect(blockObject[0].transactions).to.be.an('array');
+			expect(blockObject[0].transactions[0]).to.be.an('object');
+			expect(blockObject[0].transactions[0].id).to.equal('6950874693022090568');
+			expect(blockObject[0].transactions[0].type).to.equal(0);
 		});
 
-		describe('when block with id exists', () => {
-			describe('and block indices are duplicated', () => {
-				it('should not return duplicated blocks');
-			});
-
-			describe('and block indices are unique', () => {
-				describe('and there are no transactions in the block', () => {
-					it('should return the block containing transactions = {}');
-				});
+		it('should generate fake signature for genesis block', () => {
+			var genesisBlock_view_full_blocks_list = [
+				{
+					b_id: '6524861224470851795',
+					b_height: 1,
+					t_id: '1465651642158264047',
+					t_type: 0,
+				},
+				{
+					b_id: '6524861224470851795',
+					b_height: 1,
+					t_id: '3634383815892709956',
+					t_type: 2,
+				},
+			];
+
+			var blockObject = blocksUtilsModule.readDbRows(
+				genesisBlock_view_full_blocks_list
+			);
 
-				describe('and there are transactions in the block', () => {
-					it('should return the block containing transactions');
-				});
-			});
+			expect(blockObject).to.be.an('array');
+			expect(blockObject[0]).to.be.an('object');
+			expect(blockObject[0].id).to.equal('6524861224470851795');
+			expect(blockObject[0].generationSignature).to.equal(
+				'0000000000000000000000000000000000000000000000000000000000000000'
+			);
 		});
 	});
 
 	describe('loadBlocksPart', () => {
-		describe('when there is no error when loading block array', () => {
-			it('should be normalized');
-
-			it('should call callback with error = undefined');
-
-			it('should call callback with blocks as result');
+		it('should return error when library.db.blocks.loadBlocksData fails', done => {
+			library.db.blocks.getHeightByLastId = sinonSandbox.stub().resolves([]);
+			library.db.blocks.loadBlocksData = sinonSandbox
+				.stub()
+				.throws(new Error('An error'));
+
+			blocksUtilsModule.loadBlocksPart({}, (err, blocks) => {
+				expect(loggerStub.error.args[0][0]).to.contains('An error');
+				expect(err).to.equal('Blocks#loadBlockData error');
+				expect(blocks).to.be.undefined;
+				done();
+			});
 		});
 
-		describe('when error is defined', () => {
-			it('should call callback with the error object');
-
-			it('should call callback with blocks as result');
+		it('should return block object', done => {
+			library.db.blocks.loadBlocksData = sinonSandbox
+				.stub()
+				.resolves(viewRow_full_blocks_list);
+
+			library.db.blocks.getHeightByLastId = sinonSandbox
```

Comment:
Why we stub that function here?

Reply:
Because we are testing the child promise errors are caught.

Reply:
I don't really see a point, `loadBlocksPart` is doing following:
- call `loadBlocksData` with params
- check for error
- call `readDbRows` when there is no error
- return error if any and results if any

`library.db.blocks.getHeightByLastId` is outside of that behavior, should be testes as part of unit tests for `loadBlocksData`.

Reply:
Yes ok, will remove.

Reply:
Done.